### DISCUSSION TITLE
transcription of audio extracts from timestamps

### DISCUSSION THREAD

**Discussion by solene-evain** [openai/whisper#994] [Q&A]
Hi, 

I've got a dataset I need to transcribe. The audio files aren't cut (they are ~1h long), but I've got the timestamps according to linguistic segmentation.
The problem is: I understand that whisper takes 30 seconds extracts when the input is a long audio file. How can I make it take extracts according to timestamps, without cutting the audio file and feeding whisper with lots of little pieces of audio? 

thanks

**Comment by glangford** [comment]
> How can I make it take extracts according to timestamps, without cutting the audio file and feeding whisper with lots of little pieces of audio?

`whisper` doesn't currently have a mechanism to choose extracts according to timestamps.

What you could do is use `ffmpeg` to create extracts using the methods suggested here (I realize this is what you are trying to avoid):
* https://github.com/openai/whisper/discussions/135
* https://github.com/openai/whisper/discussions/380

Rather than creating lots of little files, one for each subsegment, you could programmatically extract each segment using `ffmpeg` and pipe it into `whisper`. Then with the `whisper` srt output, offset the timestamps to the proper location in the audio file.

**Reply by solene-evain** [reply]
Thanks! I'll try doing so :)
